src/Common/src/CoreLib/System/MemoryExtensions.cs

```diff
@@ -1304,5 +1305,36 @@ public static bool Overlaps<T>(this ReadOnlySpan<T> first, ReadOnlySpan<T> secon
                 value, comparer);
             return BinarySearch(span, comparable);
         }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool IsTypeNumeric<T>(out int size)
```

Comment:
A better name would be something like `IsComparableAsBytes`. `IsTypeNumeric` sounds like it should include types like R4 and R8 and Decimal (which it shouldn't - you can't use byte compare to compare those types) Also, `char` isn't really "numeric."

You could also test for IntPtr/UIntPtr here.

Reply:
Might be better just to have to return a `nuint` size. You're casting it to `nuint` everywhere you use it anyway.